src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -558,25 +558,76 @@ def _get_components_data(self, batteries: Set[int]) -> List[InvBatPair]:
 
             inverter_id: int = self._bat_inv_map[battery_id]
 
-            battery_data: Optional[BatteryData] = self._battery_receivers[
-                battery_id
-            ].peek()
-
-            if battery_data is None:
-                _logger.warning("None returned from battery receiver %d.", battery_id)
-                continue
-
-            inverter_data: Optional[InverterData] = self._inverter_receivers[
-                inverter_id
-            ].peek()
-
-            if inverter_data is None:
-                _logger.warning("None returned from inverter receiver %d.", inverter_id)
+            data = self._get_battery_inverter_data(battery_id, inverter_id)
+            if data is None:
+                _logger.warning(
+                    "Skipping battery %d because its message isn't correct.",
+                    battery_id,
+                )
                 continue
 
-            pairs_data.append(InvBatPair(battery_data, inverter_data))
+            pairs_data.append(data)
         return pairs_data
 
+    def _get_battery_inverter_data(
+        self, battery_id: int, inverter_id: int
+    ) -> Optional[InvBatPair]:
+        battery_data = self._battery_receivers[battery_id].peek()
+        inverter_data = self._inverter_receivers[inverter_id].peek()
+
+        # It means that nothing has been send on this channels, yet.
+        # This should be handled by BatteryStatus so it should never happen.
+        if battery_data is None or inverter_data is None:
```

Comment:
Not sure if it's worth it, but if it should never happen maybe log an error in case there is some problem that can be detected earlier.

Reply:
This moment when you get `Not sure if it's worth it` comment and you realize it is bug :D
https://github.com/frequenz-floss/frequenz-sdk-python/pull/247/commits/198f05f71a3607cfa496133813c7f79caebaf600